### PR TITLE
report_used_space probe added

### PR DESCRIPTION
**Pull request by FernandoGarzon**

### CODE REVIEW

File: cms/check_report_used_space [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+if __name__ == "__main__":
+    try:
+        status = OK
+        session = get_session()
+        for rse in list_rses():
+            limits = get_rse_usage(rse['id'])
+            for usage in limits:
+                source = usage['source']
+                attributes = list_rse_attributes(rse['id'])
```
Comment (ericvaandering) [inline cms/check_report_used_space:39]: The attributes don't change. Should be in the outer loop, not this one.

File: cms/check_report_used_space [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                if 'country' in attributes.keys():
+                    country = attributes['country']
+                else:
+                    country = 'UNKWON'
```
Comment (ericvaandering) [inline cms/check_report_used_space:43]: Spell it correctly

Comment (ericvaandering) [reply cms/check_report_used_space:43]: You can also use a dictionary "get" method to reduce these four lines to one

File: cms/check_report_used_space [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                query = session.query(models.RSE.rse_type).filter(models.RSE.id == rse['id'])
+                result = query.one()
+                type_ = result[0]
```
Comment (ericvaandering) [inline cms/check_report_used_space:46]: Good job not using type, but why not rse_type instead of type_

**Review by ericvaandering — CHANGES_REQUESTED** [review]

**Comment by ericvaandering** [comment]
@dchristidis Can you merge this one too? But I think this may be of broader use to others as well.

File: cms/check_report_used_space [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# Copyright European Organization for Nuclear Research (CERN) 2013
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Authors:
+# - Donata Mielaikaite, <donata.mielaikaite@cern.ch>, 2020
+# - Fernando Garzon, <oscar.fernando.garzon.miguez.ch> 2020
```
Comment (dchristidis) [inline cms/check_report_used_space:10]: Please replace the licence block and consistently format the authors.

File: cms/check_report_used_space [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+'''
+Probe to check used space.
+'''
+
+import datetime
+import unicodedata
```
Comment (dchristidis) [inline cms/check_report_used_space:17]: These two aren’t used.

File: cms/check_report_used_space [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import sys
+import traceback
+from rucio.core import monitor
+from rucio.core.rse import list_rses, get_rse_usage, list_rse_attributes
+from rucio.db.sqla import models
+from rucio.db.sqla.session import BASE, get_session
```
Comment (dchristidis) [inline cms/check_report_used_space:23]: `BASE` can be removed.

File: cms/check_report_used_space [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+# Exit statuses
+OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
+
+if BASE.metadata.schema:
+    schema = BASE.metadata.schema + '.'
```
Comment (dchristidis) [inline cms/check_report_used_space:29]: This is no longer needed.

File: cms/check_report_used_space [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+if __name__ == "__main__":
+    try:
+        status = OK
```
Comment (dchristidis) [inline cms/check_report_used_space:33]: `status` isn’t actually used. Consider either explicitly calling `sys.exit()` at the end or removing this.

File: cms/check_report_used_space [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        session = get_session()
+        for rse in list_rses():
+            limits = get_rse_usage(rse['id'])
```
Comment (dchristidis) [inline cms/check_report_used_space:36]: I am under the impression that the name `limits` is misleading.

File: cms/check_report_used_space [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            attributes = list_rse_attributes(rse['id'])
+            for usage in limits:
+                source = usage['source']
+                country = attributes.get('country', 'UNKNOWN')
+                query = session.query(models.RSE.rse_type).filter(models.RSE.id == rse['id'])
+                result = query.one()
+                rse_type = result[0]
```
Comment (dchristidis) [inline cms/check_report_used_space:43]: These should be moved outside of the loop: they are the same per RSE, no need to re-fetch them per RSE usage.

Comment (dchristidis) [inline cms/check_report_used_space:43]: I have a preference for using `rucio.core.rse.get_rse`, which allows you to get rid of the database-related imports. If you disagree, at least use `.scalar()` and combine everything in a single statement.

**Review by dchristidis — COMMENTED** [review]
I have reviewed your PR. Once again, my comments are mostly cosmetic.

@ericvaandering, if you don’t think they are necessary, please approve the PR and I’ll merge it immediately.

**Comment by ericvaandering** [comment]
> I have reviewed your PR. Once again, my comments are mostly cosmetic.
> 
> @ericvaandering, if you don’t think they are necessary, please approve the PR and I’ll merge it immediately.

These are all good and @FernandoGarzon should address them.

**Comment by FernandoGarzon** [comment]
@dchristidis all your comments were taken into account. Thanks.

**Comment by ericvaandering** [comment]
Looks good. Merging

**Pull request merged**